src/lib/esnext.collection.d.ts

```diff
@@ -9,3 +9,49 @@ interface MapConstructor {
         keySelector: (item: T, index: number) => K,
     ): Map<K, T[]>;
 }
+
+interface SetLike<T> {
+    /**
+     * Despite its name, returns an iterable of the values in the set-like.
+     */
+    keys(): Iterable<T>;
+    /**
+     * @returns a boolean indicating whether an element with the specified value exists in the set-like or not.
+     */
+    has(value: T): boolean;
+    /**
+     * @returns the number of (unique) elements in the set-like.
+     */
+    readonly size: number;
+}
+
+interface Set<T> {
+    /**
+     * @returns a new Set containing all the elements in this Set and also all the elements in the argument.
+     */
+    union(other: SetLike<T>): Set<T>;
+    /**
+     * @returns a new Set containing all the elements which are both in this Set and in the argument.
+     */
+    intersection(other: SetLike<T>): Set<T>;
+    /**
+     * @returns a new Set containing all the elements in this Set which are not also in the argument.
+     */
+    difference(other: SetLike<T>): Set<T>;
+    /**
+     * @returns a new Set containing all the elements in this Set which are in this or in the argument, but not in both.
+     */
+    symmetricDifference(other: SetLike<T>): Set<T>;
+    /**
+     * @returns a boolean indicating whether all the elements in this Set are also in the argument.
+     */
+    isSubsetOf(other: SetLike<T>): Set<T>;
+    /**
+     * @returns a boolean indicating whether all the elements in the argument are also in this Set.
+     */
+    isSupersetOf(other: SetLike<T>): Set<T>;

```

Comment:
These should return `boolean`s like you mentioned 😄

Reply:
lol I don't know how I managed that.